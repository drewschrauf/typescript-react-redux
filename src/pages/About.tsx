/* eslint-disable react/no-danger */
import React from 'react';
import styled from 'styled-components';

import withErrorBoundary from '@/hoc/withErrorBoundary';
import useUpdateTitle from '@/hooks/useUpdateTitle';
import readme from '../../README.md';

const ReadmeContent = styled.div`
  a {
    display: none;
  }
`;

const AboutPage: React.FC = () => {
  useUpdateTitle('About');

  return (
    <div data-testid="about-page">
      <ReadmeContent dangerouslySetInnerHTML={{ __html: readme }} />
      <p>
        <a
          href="https://github.com/drewschrauf/typescript-react-redux"
          target="_blank"
          rel="noopener noreferrer"
        >
          View on GitHub
        </a>
      </p>
    </div>
  );
};
export default withErrorBoundary()(AboutPage);
