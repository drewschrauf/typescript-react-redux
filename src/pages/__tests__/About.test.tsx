import React from 'react';
import { render } from 'react-testing-library';
import 'jest-dom/extend-expect';

import About from '../About';

jest.mock('../../../README.md', () => '<h1>Content</h1>');

describe('About', () => {
  it('should match snapshot', () => {
    const root = render(<About />);
    expect(root.container.innerHTML).toMatchSnapshot();
  });

  it('should render README content as markup', () => {
    const root = render(<About />);
    expect(root.getByText('Content')).toBeInTheDocument();
  });
});
