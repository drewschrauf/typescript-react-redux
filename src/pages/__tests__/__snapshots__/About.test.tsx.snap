// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match snapshot 1`] = `"<div><div data-test=\\"content\\" class=\\"sc-bdVaJa fsIviB\\"><h1>Content</h1></div><p><a href=\\"https://github.com/drewschrauf/typescript-react-redux\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">View on GitHub</a></p></div>"`;
